### PR TITLE
Fix RIGHT / FULL correlated join decorrelation

### PR DESCRIPTION
**Pull request by kasiafi**

### CODE REVIEW

**Comment by kasiafi** [comment]
Added tests.

**Comment by kasiafi** [comment]
Added unit test for the rule.

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java [hunk 1]
```diff
@@ -64,27 +77,98 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
...
+                    correlatedJoinNode.getFilter());
+            return Result.ofPlanNode(rewriteToJoin(
+                    correlatedJoinNode,
+                    correlatedJoinNode.getType().toJoinNodeType(),
+                    filter,
+                    decorrelatedSubquery.getNode()));
+        }
+
+        checkState(
+                correlatedJoinNode.getType() == RIGHT || correlatedJoinNode.getType() == FULL,
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java]: do we expect other kinds of joins in the future?

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java [hunk 2]
```diff
@@ -15,21 +15,34 @@
 
 import com.google.common.collect.ImmutableList;
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java]: could you add more description to commit message? what does it fix specifically?

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java (hunk 1 above)
Comment (kasiafi) [reply presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java]: I don't think we do, yet I think it should be explicit what types are handled where.

**Comment by kasiafi** [comment]
Added explanation of the change in the commit message.

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java (hunk 1 above)
Comment (martint) [reply presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java]: @sopel39, this is similar to what needs to be done to ensure that all branches of a switch are exhaustive.

**Comment by kasiafi** [comment]
Please do not merge. There's one corner case missed. I will fix it soon.

**Comment by kasiafi** [comment]
Fixed incorrect decorrelation in the case of empty CorrelatedJoin's input.

**Comment by ebyhr** [comment]
@kasiafi Could you fix some code style?
```sh
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[103] (indentation) Indentation: 'method def' child has incorrect indentation level 12, expected level should be 8.
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[104] (indentation) Indentation: 'if' has incorrect indentation level 12, expected level should be 8.
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[105] (indentation) Indentation: 'if' child has incorrect indentation level 16, expected level should be 12.
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[106] (indentation) Indentation: 'if rcurly' has incorrect indentation level 12, expected level should be 8.
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[107] (indentation) Indentation: 'else' has incorrect indentation level 12, expected level should be 8.
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[108] (indentation) Indentation: 'else' child has incorrect indentation level 16, expected level should be 12.
[ERROR] src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedJoinToJoin.java:[109] (indentation) Indentation: 'else rcurly' has incorrect indentation level 12, expected level should be 8.
```

**Pull request merged**